src/sentry/tasks/relay.py

```diff
@@ -119,26 +119,26 @@ def compute_configs(organization_id=None, project_id=None, public_key=None):
         # some point this should be handled better.
         projects = list(Project.objects.filter(organization_id=organization_id))
         for key in ProjectKey.objects.filter(project__in=projects):
-            configs[key.public_key] = None
+            if key.status == ProjectKeyStatus.ACTIVE:
+                configs[key.public_key] = None
+            else:
+                configs[key.public_key] = {"disabled": True}
     elif project_id:
         for key in ProjectKey.objects.filter(project_id=project_id):
             configs[key.public_key] = compute_projectkey_config(key)
     elif public_key:
         try:
             key = ProjectKey.objects.get(public_key=public_key)
         except ProjectKey.DoesNotExist:
-            # There are two main reasons this might happen:
+            # The invalidation task was triggered for a deletion and the
+            # ProjectKey should be deleted from the cache.
             #
-            # - The invalidation task was triggered for a deletion and the ProjectKey should
-            #   be deleted from the cache.
-            # - Django fired the `after_save` event before a transaction creating the
-            #   ProjectKey was committed.
-            #
-            # Thus we want to make sure we delete the project, but we do not care about
-            # disabling it here, because doing so would cause it to be wrongly disabled for
-            # an hour in the second case (which will be fixed at some point).  If the v3
-            # task finds a non-existing ProjectKey it can disable this project.
-            configs[public_key] = None
+            # This used to delete the cache entry instead of disabling it. The
+            # reason for that was to work around a bug in our model signal
+            # handlers that sent off the invalidation tasks before the DB
+            # transaction was committed, causing us to write stale caches. That
+            # bug was fixed in https://github.com/getsentry/sentry/pull/35671
+            configs[public_key] = {"disabled": True}
```

Comment:
Not sure if this is a concern, but could somebody use this to spam our redis cache with made up project keys?

Reply:
I had to think about this for a while, but generally we have IP address rate limits in the load balancer for this kind of thing